File: ExcelLikeTextAlignment_CS/Form1.cs
using System.Windows.Forms;

namespace ExcelLikeTextAlignment_CS
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
            gridControl1.RowCount = 10;
            gridControl1.ColCount = 6;
            gridControl1.ExcelLikeAlignment = true;
            //Fill first half of columns with numeric values
            for (int i = 1; i <= gridControl1.RowCount; i++)
            {
                for (int j = 0; j <= gridControl1.ColCount/2; j++)
                {
                    gridControl1[i, j].CellValue = i+j;
                }
            }
            //Fill second half of columns with "Data"
            for (int i = 1; i <= gridControl1.RowCount; i++)
            {
                for (int j = gridControl1.ColCount / 2+1; j <= gridControl1.ColCount; j++)
                {
                    gridControl1[i, j].CellValue = "Data";
                }
            }
        }
    }
}

File: ExcelLikeTextAlignment_VB/Form1.vb
Public Class Form1
    Public Sub New()
        InitializeComponent()

        Me.GridControl1.RowCount = 10
        Me.GridControl1.ColCount = 6
        Me.GridControl1.ExcelLikeAlignment = True

        ' Fill first half of columns with numeric values
        For i As Integer = 1 To GridControl1.RowCount
            For j As Integer = 0 To GridControl1.ColCount \ 2
                Me.GridControl1(i, j).CellValue = i + j
            Next
        Next

        ' Fill second half of columns with "Data"
        For i As Integer = 1 To GridControl1.RowCount
            For j As Integer = (GridControl1.ColCount \ 2) + 1 To GridControl1.ColCount
                Me.GridControl1(i, j).CellValue = "Data"
            Next
        Next
    End Sub
End Class